client/go/util/operationResult.go

```diff
@@ -0,0 +1,32 @@
+// Copyright Verizon Media. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+// A struct containing the result of an operation
+// Author: bratseth
+
+package util
+
+type OperationResult struct {
+	Success bool
+	Message string // Mandatory message
+	Detail  string // Optional detail message
+	Payload string // Optional payload - may be present whether or not the operation was success
+}
+
+func Success(message string) *OperationResult {
```

Comment:
I think it's better to return a copy here (and the other constructors), seeing as this a light-weight struct.